src/main/java/org/opentripplanner/graph_builder/module/time/TrafifcPredictionBuilderModule.java

```diff
@@ -0,0 +1,39 @@
+package org.opentripplanner.graph_builder.module.time;
+
+import org.opentripplanner.graph_builder.services.GraphBuilderModule;
+import org.opentripplanner.routing.edgetype.StreetEdge;
+import org.opentripplanner.routing.graph.Graph;
+
+import java.io.File;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.TreeMap;
+
+
+public class TrafifcPredictionBuilderModule implements GraphBuilderModule {
+    private final ClusterList clusterlist;
+
+    public TrafifcPredictionBuilderModule(File traficprediction) {
+        this.clusterlist = new ClusterList(traficprediction);
+        this.clusterlist.getclusters().sort(Comparator.naturalOrder());
+    }
+
+    @Override
+    public void buildGraph(Graph graph, HashMap<Class<?>, Object> extra) {
+        graph.setClusters(this.clusterlist);
+        TreeMap<Integer, Integer> map = new TreeMap<>();
+        for (Cluster c : graph.getClusters().getclusters()) {
+            for (EdgeData e : c.getedges())
+                map.put(((int) e.getid()), e.getclusterid());
+        }
+        for (StreetEdge e : graph.getStreetEdges()) {
+            e.setTimes(this.clusterlist.getclusters().get(map.get(e.getId())).gettimetableas());
```

Comment:
Tutaj `map.get(e.getId())` może zwrócić nulla - trzeba to wyifować. Swoją drogą, to ta mapa `map` mogłaby być od razu z identyfikacji wierzchołka w timetable.

Reply:
fixed